src/cmd-build

```diff
@@ -227,23 +228,36 @@ else
     extraargs="${extraargs} --kickstart ${image_input}"
 fi
 
-# This generates the "base image"; not specific to a platform.
-# We want extraargs var to be split on words
-# shellcheck disable=SC2086
-/usr/lib/coreos-assembler/virt-install --dest="${PWD}"/"${img_base}" \
-               --create-disk --kickstart-out "${PWD}"/tmp/flattened.ks \
-               --ostree-remote="${name}" --ostree-stateroot="${name}" \
-               --ostree-ref="${ref:-${commit}}" \
-               --location "${workdir}"/installer/*.iso \
-               --ostree-repo="${workdir}"/repo ${extraargs-}
-/usr/lib/coreos-assembler/gf-anaconda-cleanup "$(pwd)"/"${img_base}"
 set +x
+# This generates the "base image"; not specific to a platform.
+run_virtinstall() {
+    dest=$1
+    shift
+    # We want extraargs var to be split on words
+    # shellcheck disable=SC2086
+    /usr/lib/coreos-assembler/virt-install --create-disk --dest=${dest} \
+                                           --kickstart-out "${PWD}"/tmp/flattened.ks \
+                                           --ostree-remote="${name}" --ostree-stateroot="${name}" \
+                                           --ostree-ref="${ref:-${commit}}" \
+                                           --location "${workdir}"/installer/*.iso \
+                                           --ostree-repo="${workdir}"/repo ${extraargs-} "$@"
+    # And strip out all of the Anaconda stuff in /var
+    /usr/lib/coreos-assembler/gf-anaconda-cleanup "${dest}"
+}
+
+build_cloud_base() {
+    if [ -f "$(pwd)/${img_base}" ]; then
+        return
```

Comment:
so if it exists we assume it's good? looks like we just re-ran virt-install in the past. Just wondering if this will have any side effects.

Reply:
this looks good. thanks

Reply:
did you make any changes here? looks the same?

Reply:
See https://github.com/coreos/coreos-assembler/pull/305/commits/8a5f155b92de2cf7d1585aca310d8e893f8d5991#diff-67dd78a4a0065a3a3ce11c79d9fc7117R236

Reply:
👍